### ISSUE TITLE
Warning: Cannot define property:48, object is not extensible. Use --force to continue.

### ISSUE THREAD

**Issue by inlikealion** [yeoman/generator-webapp_DEPRECATED#232]
Fresh update `npm update -g` then `mkdir mywebapp && yo webapp`, followed by `grunt`:

```
Running "mocha:all" (mocha) task
Testing: http://127.0.0.1:9001/index.html
$ Warning: Cannot define property:48, object is not extensible. Use --force to continue.
```

**Comment by eddiemonge** [comment]
I am not seeing that at all. What OS?

**Comment by inlikealion** [comment]
OS X 10.8.5.

**Comment by eddiemonge** [comment]
can you post your code? You didn't change anything at all?

**Comment by inlikealion** [comment]
Tell me what code you want. As demonstrated above I have got fresh `npm` updates and done an initial `yo webapp` in a new, blank directory, followed directly by the failed `grunt` task.

**Comment by addyosmani** [comment]
Haven't been able to reproduce this on Maverick, but thanks for the issue report! Could you share the Gruntfile you're using for this project? If nothing has been changed at all it's possible that either a) an older version is (somehow) still being used despite the update being made or b) a dependency has failed to correctly install causing some breakage.

We'll do our best to help!

**Comment by inlikealion** [comment]
```
// Generated on 2013-12-17 using generator-webapp 0.4.6
'use strict';

// # Globbing
// for performance reasons we're only matching one level down:
// 'test/spec/{,*/}*.js'
// use this if you want to recursively match all subfolders:
// 'test/spec/**/*.js'

module.exports = function (grunt) {

    // Load grunt tasks automatically
    require('load-grunt-tasks')(grunt);

    // Time how long tasks take. Can help when optimizing build times
    require('time-grunt')(grunt);

    // Define the configuration for all the tasks
    grunt.initConfig({

        // Project settings
        yeoman: {
            // Configurable paths
            app: 'app',
            dist: 'dist'
        },

        // Watches files for changes and runs tasks based on the changed files
        watch: {
            js: {
                files: ['<%= yeoman.app %>/scripts/{,*/}*.js'],
                tasks: ['jshint'],
                options: {
                    livereload: true
                }
            },
            jstest: {
                files: ['test/spec/{,*/}*.js'],
                tasks: ['test:watch']
            },
            gruntfile: {
                files: ['Gruntfile.js']
            },
            compass: {
                files: ['<%= yeoman.app %>/styles/{,*/}*.{scss,sass}'],
                tasks: ['compass:server', 'autoprefixer']
            },
            styles: {
                files: ['<%= yeoman.app %>/styles/{,*/}*.css'],
                tasks: ['newer:copy:styles', 'autoprefixer']
            },
            livereload: {
                options: {
                    livereload: '<%= connect.options.livereload %>'
                },
                files: [
                    '<%= yeoman.app %>/{,*/}*.html',
                    '.tmp/styles/{,*/}*.css',
                    '<%= yeoman.app %>/images/{,*/}*.{gif,jpeg,jpg,png,svg,webp}'
                ]
            }
        },

        // The actual grunt server settings
        connect: {
            options: {
                port: 9000,
                livereload: 35729,
                // Change this to '0.0.0.0' to access the server from outside
                hostname: 'localhost'
            },
            livereload: {
                options: {
                    open: true,
                    base: [
                        '.tmp',
                        '<%= yeoman.app %>'
                    ]
                }
            },
            test: {
                options: {
                    port: 9001,
                    base: [
                        '.tmp',
                        'test',
                        '<%= yeoman.app %>'
                    ]
                }
            },
            dist: {
                options: {
                    open: true,
                    base: '<%= yeoman.dist %>',
                    livereload: false
                }
            }
        },

        // Empties folders to start fresh
        clean: {
            dist: {
                files: [{
                    dot: true,
                    src: [
                        '.tmp',
                        '<%= yeoman.dist %>/*',
                        '!<%= yeoman.dist %>/.git*'
                    ]
                }]
            },
            server: '.tmp'
        },

        // Make sure code styles are up to par and there are no obvious mistakes
        jshint: {
            options: {
                jshintrc: '.jshintrc',
                reporter: require('jshint-stylish')
            },
            all: [
                'Gruntfile.js',
                '<%= yeoman.app %>/scripts/{,*/}*.js',
                '!<%= yeoman.app %>/scripts/vendor/*',
                'test/spec/{,*/}*.js'
            ]
        },


        // Mocha testing framework configuration options
        mocha: {
            all: {
                options: {
                    run: true,
                    urls: ['http://<%= connect.test.options.hostname %>:<%= connect.test.options.port %>/index.html']
                }
            }
        },



        // Compiles Sass to CSS and generates necessary files if requested
        compass: {
            options: {
                sassDir: '<%= yeoman.app %>/styles',
                cssDir: '.tmp/styles',
                generatedImagesDir: '.tmp/images/generated',
                imagesDir: '<%= yeoman.app %>/images',
                javascriptsDir: '<%= yeoman.app %>/scripts',
                fontsDir: '<%= yeoman.app %>/styles/fonts',
                importPath: '<%= yeoman.app %>/bower_components',
                httpImagesPath: '/images',
                httpGeneratedImagesPath: '/images/generated',
                httpFontsPath: '/styles/fonts',
                relativeAssets: false,
                assetCacheBuster: false
            },
            dist: {
                options: {
                    generatedImagesDir: '<%= yeoman.dist %>/images/generated'
                }
            },
            server: {
                options: {
                    debugInfo: true
                }
            }
        },

        // Add vendor prefixed styles
        autoprefixer: {
            options: {
                browsers: ['last 1 version']
            },
            dist: {
                files: [{
                    expand: true,
                    cwd: '.tmp/styles/',
                    src: '{,*/}*.css',
                    dest: '.tmp/styles/'
                }]
            }
        },

        // Automatically inject Bower components into the HTML file
        'bower-install': {
            app: {
                html: '<%= yeoman.app %>/index.html',
                ignorePath: '<%= yeoman.app %>/'
            }
        },

        // Renames files for browser caching purposes
        rev: {
            dist: {
                files: {
                    src: [
                        '<%= yeoman.dist %>/scripts/{,*/}*.js',
                        '<%= yeoman.dist %>/styles/{,*/}*.css',
                        '<%= yeoman.dist %>/images/{,*/}*.{gif,jpeg,jpg,png,webp}',
                        '<%= yeoman.dist %>/styles/fonts/{,*/}*.*'
                    ]
                }
            }
        },

        // Reads HTML for usemin blocks to enable smart builds that automatically
        // concat, minify and revision files. Creates configurations in memory so
        // additional tasks can operate on them
        useminPrepare: {
            options: {
                dest: '<%= yeoman.dist %>'
            },
            html: '<%= yeoman.app %>/index.html'
        },

        // Performs rewrites based on rev and the useminPrepare configuration
        usemin: {
            options: {
                assetsDirs: ['<%= yeoman.dist %>']
            },
            html: ['<%= yeoman.dist %>/{,*/}*.html'],
            css: ['<%= yeoman.dist %>/styles/{,*/}*.css']
        },

        // The following *-min tasks produce minified files in the dist folder
        imagemin: {
            dist: {
                files: [{
                    expand: true,
                    cwd: '<%= yeoman.app %>/images',
                    src: '{,*/}*.{gif,jpeg,jpg,png}',
                    dest: '<%= yeoman.dist %>/images'
                }]
            }
        },
        svgmin: {
            dist: {
                files: [{
                    expand: true,
                    cwd: '<%= yeoman.app %>/images',
                    src: '{,*/}*.svg',
                    dest: '<%= yeoman.dist %>/images'
                }]
            }
        },
        htmlmin: {
            dist: {
                options: {
                    collapseBooleanAttributes: true,
                    collapseWhitespace: true,
                    removeAttributeQuotes: true,
                    removeCommentsFromCDATA: true,
                    removeEmptyAttributes: true,
                    removeOptionalTags: true,
                    removeRedundantAttributes: true,
                    useShortDoctype: true
                },
                files: [{
                    expand: true,
                    cwd: '<%= yeoman.dist %>',
                    src: '{,*/}*.html',
                    dest: '<%= yeoman.dist %>'
                }]
            }
        },

        // By default, your `index.html`'s <!-- Usemin block --> will take care of
        // minification. These next options are pre-configured if you do not wish
        // to use the Usemin blocks.
        // cssmin: {
        //     dist: {
        //         files: {
        //             '<%= yeoman.dist %>/styles/main.css': [
        //                 '.tmp/styles/{,*/}*.css',
        //                 '<%= yeoman.app %>/styles/{,*/}*.css'
        //             ]
        //         }
        //     }
        // },
        // uglify: {
        //     dist: {
        //         files: {
        //             '<%= yeoman.dist %>/scripts/scripts.js': [
        //                 '<%= yeoman.dist %>/scripts/scripts.js'
        //             ]
        //         }
        //     }
        // },
        // concat: {
        //     dist: {}
        // },

        // Copies remaining files to places other tasks can use
        copy: {
            dist: {
                files: [{
                    expand: true,
                    dot: true,
                    cwd: '<%= yeoman.app %>',
                    dest: '<%= yeoman.dist %>',
                    src: [
                        '*.{ico,png,txt}',
                        '.htaccess',
                        'images/{,*/}*.webp',
                        '{,*/}*.html',
                        'styles/fonts/{,*/}*.*',
                        'bower_components/sass-bootstrap/fonts/*.*'
                    ]
                }]
            },
            styles: {
                expand: true,
                dot: true,
                cwd: '<%= yeoman.app %>/styles',
                dest: '.tmp/styles/',
                src: '{,*/}*.css'
            }
        },


        // Generates a custom Modernizr build that includes only the tests you
        // reference in your app
        modernizr: {
            devFile: '<%= yeoman.app %>/bower_components/modernizr/modernizr.js',
            outputFile: '<%= yeoman.dist %>/bower_components/modernizr/modernizr.js',
            files: [
                '<%= yeoman.dist %>/scripts/{,*/}*.js',
                '<%= yeoman.dist %>/styles/{,*/}*.css',
                '!<%= yeoman.dist %>/scripts/vendor/*'
            ],
            uglify: true
        },

        // Run some tasks in parallel to speed up build process
        concurrent: {
            server: [
                'compass:server',
                'copy:styles'
            ],
            test: [
                'copy:styles'
            ],
            dist: [
                'compass',
                'copy:styles',
                'imagemin',
                'svgmin'
            ]
        }
    });


    grunt.registerTask('serve', function (target) {
        if (target === 'dist') {
            return grunt.task.run(['build', 'connect:dist:keepalive']);
        }

        grunt.task.run([
            'clean:server',
            'concurrent:server',
            'autoprefixer',
            'connect:livereload',
            'watch'
        ]);
    });

    grunt.registerTask('server', function () {
        grunt.log.warn('The `server` task has been deprecated. Use `grunt serve` to start a server.');
        grunt.task.run(['serve']);
    });

    grunt.registerTask('test', function(target) {
        if (target !== 'watch') {
            grunt.task.run([
                'clean:server',
                'concurrent:test',
                'autoprefixer',
            ]);
        }

        grunt.task.run([
            'connect:test',
            'mocha'
        ]);
    });

    grunt.registerTask('build', [
        'clean:dist',
        'useminPrepare',
        'concurrent:dist',
        'autoprefixer',
        'concat',
        'cssmin',
        'uglify',
        'copy:dist',
        'modernizr',
        'rev',
        'usemin',
        'htmlmin'
    ]);

    grunt.registerTask('default', [
        'newer:jshint',
        'test',
        'build'
    ]);
};
```

**Comment by eddiemonge** [comment]
Can you try the following please?

```
npm remove -g yo generator-webapp bower
npm cache clean
npm install -g yo generator-webapp
mkdir mywebapp
yo webapp
[enter enter]
grunt test
grunt
```

**Comment by inlikealion** [comment]
Yes sir. Will do and report back.

**Comment by inlikealion** [comment]
```
npm remove -g yo generator-webapp bower
npm cache clean
npm install -g yo generator-webapp
mkdir mywebapp && cd mywebapp
yo webapp
[enter enter]
grunt test
grunt
```

Updated your directions in case someone less familiar with the Terminal lands here and doesn’t realize they need to run `yo` from inside their fresh project directory.

**Comment by inlikealion** [comment]
Failed on `grunt test` with:

```
Running "test" task

Running "clean:server" (clean) task

Running "concurrent:test" (concurrent) task

    Running "copy:styles" (copy) task


    Done, without errors.

Running "autoprefixer:dist" (autoprefixer) task

Running "connect:test" (connect) task
Started connect web server on 127.0.0.1:9001.

Running "mocha:all" (mocha) task
Testing: http://127.0.0.1:9001/index.html
Warning: Cannot define property:48, object is not extensible. Use --force to continue.

Aborted due to warnings.


Execution Time (2013-12-17 18:28:15 UTC)
concurrent:test     3s  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 87%
connect:test     347ms  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 11%
Total 3s
```

**Comment by eddiemonge** [comment]
Do you have mocha installed globally already? Can you run `grunt test --verbose`?

**Comment by inlikealion** [comment]
I haven’t installed mocha on its own globally outside of what the yeoman install process does. I assumed that the generator would get all its own dependencies or things needed would be noted in the documentation or error output (like generators’ relience on Ruby, Sass, or Compass).

**Comment by inlikealion** [comment]
`grunt test --verbose` outputs:

```
Initializing
Command-line options: --verbose

Reading "Gruntfile.js" Gruntfile...OK

Registering Gruntfile tasks.

Registering "grunt-contrib-copy" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-copy/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-copy/package.json...OK
Loading "copy.js" tasks...OK
+ copy

Registering "grunt-contrib-concat" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-concat/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-concat/package.json...OK
Loading "concat.js" tasks...OK
+ concat

Registering "grunt-contrib-uglify" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-uglify/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-uglify/package.json...OK
Loading "uglify.js" tasks...OK
+ uglify

Registering "grunt-contrib-compass" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-compass/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-compass/package.json...OK
Loading "compass.js" tasks...OK
+ compass

Registering "grunt-contrib-jshint" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-jshint/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-jshint/package.json...OK
Loading "jshint.js" tasks...OK
+ jshint

Registering "grunt-contrib-cssmin" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-cssmin/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-cssmin/package.json...OK
Loading "cssmin.js" tasks...OK
+ cssmin

Registering "grunt-contrib-connect" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-connect/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-connect/package.json...OK
Loading "connect.js" tasks...OK
+ connect

Registering "grunt-contrib-clean" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-clean/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-clean/package.json...OK
Loading "clean.js" tasks...OK
+ clean

Registering "grunt-contrib-htmlmin" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-htmlmin/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-htmlmin/package.json...OK
Loading "htmlmin.js" tasks...OK
+ htmlmin

Registering "grunt-bower-install" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-bower-install/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-bower-install/package.json...OK
Loading "bower-install.js" tasks...OK
+ bower-install

Registering "grunt-contrib-imagemin" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-imagemin/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-imagemin/package.json...OK
Loading "imagemin.js" tasks...OK
+ imagemin

Registering "grunt-contrib-watch" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-watch/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-watch/package.json...OK
Loading "watch.js" tasks...OK
+ watch

Registering "grunt-rev" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-rev/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-rev/package.json...OK
Loading "rev.js" tasks...OK
+ rev

Registering "grunt-autoprefixer" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-autoprefixer/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-autoprefixer/package.json...OK
Loading "autoprefixer.js" tasks...OK
+ autoprefixer

Registering "grunt-usemin" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-usemin/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-usemin/package.json...OK
Loading "usemin.js" tasks...OK
+ usemin, useminPrepare

Registering "grunt-mocha" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-mocha/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-mocha/package.json...OK
Growl not found, 'npm install growl' for Growl supportLoading "mocha.js" tasks...OK
+ mocha

Registering "grunt-modernizr" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-modernizr/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-modernizr/package.json...OK
Loading "modernizr.js" tasks...OK
+ modernizr

Registering "grunt-newer" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-newer/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-newer/package.json...OK
Loading "newer.js" tasks...OK
+ any-newer, newer, newer-postrun

Registering "grunt-svgmin" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-svgmin/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-svgmin/package.json...OK
Loading "svgmin.js" tasks...OK
+ svgmin

Registering "grunt-concurrent" local Npm module tasks.
Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-concurrent/package.json...OK
Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-concurrent/package.json...OK
Loading "concurrent.js" tasks...OK
+ concurrent
Initializing config...OK
Loading "Gruntfile.js" tasks...OK
+ build, default, serve, server, test

Running tasks: test

Running "test" task

Running "clean:server" (clean) task
Verifying property clean.server exists in config...OK
Files: [no src] -> server
Options: force=false, no-write=false
Options: force=false, no-write=false

Running "concurrent:test" (concurrent) task
Verifying property concurrent.test exists in config...OK
Files: [no src] -> test
Options: limit=4

    Initializing
    Command-line options: --verbose, --_modernizr.defaults=[object Object], --_modernizr.private=[object Object]

    Reading "Gruntfile.js" Gruntfile...OK

    Registering Gruntfile tasks.

    Registering "grunt-contrib-copy" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-copy/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-copy/package.json...OK
    Loading "copy.js" tasks...OK
    + copy

    Registering "grunt-contrib-concat" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-concat/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-concat/package.json...OK
    Loading "concat.js" tasks...OK
    + concat

    Registering "grunt-contrib-uglify" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-uglify/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-uglify/package.json...OK
    Loading "uglify.js" tasks...OK
    + uglify

    Registering "grunt-contrib-compass" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-compass/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-compass/package.json...OK
    Loading "compass.js" tasks...OK
    + compass

    Registering "grunt-contrib-jshint" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-jshint/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-jshint/package.json...OK
    Loading "jshint.js" tasks...OK
    + jshint

    Registering "grunt-contrib-cssmin" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-cssmin/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-cssmin/package.json...OK
    Loading "cssmin.js" tasks...OK
    + cssmin

    Registering "grunt-contrib-connect" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-connect/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-connect/package.json...OK
    Loading "connect.js" tasks...OK
    + connect

    Registering "grunt-contrib-clean" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-clean/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-clean/package.json...OK
    Loading "clean.js" tasks...OK
    + clean

    Registering "grunt-contrib-htmlmin" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-htmlmin/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-htmlmin/package.json...OK
    Loading "htmlmin.js" tasks...OK
    + htmlmin

    Registering "grunt-bower-install" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-bower-install/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-bower-install/package.json...OK
    Loading "bower-install.js" tasks...OK
    + bower-install

    Registering "grunt-contrib-imagemin" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-imagemin/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-imagemin/package.json...OK
    Loading "imagemin.js" tasks...OK
    + imagemin

    Registering "grunt-contrib-watch" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-watch/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-contrib-watch/package.json...OK
    Loading "watch.js" tasks...OK
    + watch

    Registering "grunt-rev" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-rev/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-rev/package.json...OK
    Loading "rev.js" tasks...OK
    + rev

    Registering "grunt-autoprefixer" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-autoprefixer/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-autoprefixer/package.json...OK
    Loading "autoprefixer.js" tasks...OK
    + autoprefixer

    Registering "grunt-usemin" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-usemin/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-usemin/package.json...OK
    Loading "usemin.js" tasks...OK
    + usemin, useminPrepare

    Registering "grunt-mocha" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-mocha/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-mocha/package.json...OK
    Growl not found, 'npm install growl' for Growl supportLoading "mocha.js" tasks...OK
    + mocha

    Registering "grunt-modernizr" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-modernizr/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-modernizr/package.json...OK
    Loading "modernizr.js" tasks...OK
    + modernizr

    Registering "grunt-newer" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-newer/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-newer/package.json...OK
    Loading "newer.js" tasks...OK
    + any-newer, newer, newer-postrun

    Registering "grunt-svgmin" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-svgmin/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-svgmin/package.json...OK
    Loading "svgmin.js" tasks...OK
    + svgmin

    Registering "grunt-concurrent" local Npm module tasks.
    Reading /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-concurrent/package.json...OK
    Parsing /Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-concurrent/package.json...OK
    Loading "concurrent.js" tasks...OK
    + concurrent
    Initializing config...OK
    Loading "Gruntfile.js" tasks...OK
    + build, default, serve, server, test

    Running tasks: copy:styles

    Running "copy:styles" (copy) task
    Verifying property copy.styles exists in config...OK
    Options: processContent=false, processContentExclude=[]
    Options: processContent=false, processContentExclude=[]


    Done, without errors.

Running "autoprefixer" task

Running "autoprefixer:dist" (autoprefixer) task
Verifying property autoprefixer.dist exists in config...OK
Options: diff=false, browsers=["last 1 version"]

Running "connect:test" (connect) task
Verifying property connect.test exists in config...OK
File: [no files]
Options: protocol="http", port=9001, hostname="localhost", base=[".tmp","test","app"], directory=null, keepalive=false, debug=false, livereload=35729, open=false, middleware=undefined
Started connect web server on 127.0.0.1:9001.

Running "mocha" task

Running "mocha:all" (mocha) task
Verifying property mocha.all exists in config...OK
File: [no files]
Options: log=false, reporter="Dot", timeout=5000, inject="/Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-mocha/phantomjs/bridge.js", phantomScript="/Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-mocha/phantomjs/main.js", urls=["http://127.0.0.1:9001/index.html"], bail=false, run
Options: {
  "log": false,
  "reporter": "Dot",
  "timeout": 5000,
  "inject": "/Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-mocha/phantomjs/bridge.js",
  "phantomScript": "/Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-mocha/phantomjs/main.js",
  "urls": [
    "http://127.0.0.1:9001/index.html"
  ],
  "bail": false,
  "run": true
}
Phantom options: {
  "timeout": 5000,
  "inject": "/Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-mocha/phantomjs/bridge.js",
  "phantomScript": "/Users/matt.bainton/Desktop/mywebapp/node_modules/grunt-mocha/phantomjs/main.js",
  "run": true
}
Testing: http://127.0.0.1:9001/index.html
Warning: Cannot define property:48, object is not extensible. Use --force to continue.

Aborted due to warnings.


Execution Time (2013-12-17 18:46:45 UTC)
loading tasks       5ms  0%
test                1ms  0%
clean:server        5ms  0%
concurrent:test      3s  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 98%
autoprefixer        1ms  0%
autoprefixer:dist  22ms  ▇▇ 1%
connect:test       22ms  ▇▇ 1%
mocha               0ms  0%
mocha:all          10ms  0%
Total 3s
```

**Comment by eddiemonge** [comment]
Ok what about `node_modules/.bin/mocha test`

**Comment by inlikealion** [comment]
Is that a project-relative directory or a project-relative command to run?

**Comment by eddiemonge** [comment]
From your project's root directory, run
`node_modules/grunt-mocha/node_modules/.bin/mocha test --recursive`

I thought it would put it in the main modules but I was wrong.

**Comment by inlikealion** [comment]
Ok, that gave us:

```
$ node_modules/grunt-mocha/node_modules/.bin/mocha test --recursive

  ․

  1 passing (9ms)
```

**Comment by eddiemonge** [comment]
and grunt test still fails?

**Comment by eddiemonge** [comment]
what version of node are you running?

**Comment by inlikealion** [comment]
You know, that could be it. I installed from git and assumed it would automatically `make` from the latest stable release (using [this method](https://gist.github.com/inlikealion/7935203)). Instead it makes from tip, so I am on `node v0.11.10-pre`. I realize this is outside the scope of the generator issue, but do you know if I can checkout a point release and re-run the `make` command to roll back?

**Comment by eddiemonge** [comment]
no idea. i know that node 0.11 does have issues and often doesn't work and thats why yeoman only official supports 0.10 and sometimes 0.8

**Comment by inlikealion** [comment]
Ok, that was it: generator-webapp and yeoman don’t play nice with `node v0.11.x`. I updated [my gist](https://gist.github.com/inlikealion/7935203) with steps for checking out a specific tag (in this case the latest on the `v0.10.x` release track) and ran `make install -j` again. Then I ran through @eddiemonge [troubleshooting tasks](https://github.com/yeoman/generator-webapp/issues/232#issuecomment-30776404) and everything is peachy keen.

Thanks for helping me track this down guys!
